### PR TITLE
Renames `entity_bundle` to `islandora_entity_bundle` to avoid name collision

### PR DESCRIPTION
**Pull request by dannylamb**
**JIRA Ticket**: Islandora/documentation#1470

Supersedes https://github.com/Islandora/islandora/pull/771

# What does this Pull Request do?

Changes the plugin_id for the entity_bundle Condition to islandora_entity_bundle to avoid a name collision with ctools entity_bundle plugin.

It also has an update hook to update any contexts that may be using the `entity_bundle` condition.

When this patch is applied, it should be possible to install pathauto and ctools and generate url aliases for taxonomy terms in the corporate_body vocabulary.

# How should this be tested?

1. On a fresh install, download & enable pathauto (also brings ctools).
1. Visit /admin/config/search/path/patterns/add and try and specify a pathauto pattern for a corporate_body term, e.g. pattern=`/agent/[term:name]`. Observe "The website encountered an unexpected error. Please try again later." or "Drupal\Component\Plugin\Exception\PluginNotFoundException: The "entity_bundle:taxonomy_term" plugin does not exist." errors.
1. Go visit `admin/structure/context` and get a whitescreen
1. Apply this PR.
1. Run the update hook by executing `drush updatedb` from the command line.
1. Re-try defining pathauto pattern. This time creating a pathauto pattern should succeed.
1. Go visit `admin/structure/context`.  You should not whitescreen

# Interested parties
@Islandora/8-x-committers

### CODE REVIEW

**Comment by dannylamb** [comment]
@kayakr Can you confirm this still works for you?

@Islandora/8-x-committers Anyone want to attempt running this update hook to make sure we're good?

**Comment by dannylamb** [comment]
Ok, that last commit should make travis happy

**Comment by dannylamb** [comment]
:tada: Travis is happy!

**Comment by kayakr** [comment]
@dannylamb I realise the branch I built the original patch in (#771) had a commit that's already been applied to 8.x-1.x (for derivative term), so I removed that.

Now the patch applied and fixes the issue.

The db update hook fails.
```
 ----------- ----------- --------------- ------------------------------------- 
  Module      Update ID   Type            Description                          
 ----------- ----------- --------------- ------------------------------------- 
  islandora   8002        hook_update_n   Replaces 'entity_bundle' conditions  
                                          with 'islandora_entity_bundle'.      
                                          This prevents plugin naming          
                                          collisions between islandora and     
                                          ctools.                              
 ----------- ----------- --------------- ------------------------------------- 

 // Do you wish to run the specified pending updates?: yes.                                                             

>  [notice] Update started: islandora_update_8002
>  [error]  You have requested a non-existent service "plugin.manager.context". 
>  [error]  Update failed: islandora_update_8002 
```
Should this be `plugin.manager.context_reaction` instead?

Do we still need a matching patch for the context definition in islandora_defaults.

**Comment by dannylamb** [comment]
Yes, we will need a matching `islandora_defaults` PR.  I can whip that up.

Not sure what's up with that service not being present tho.  I was testing in pieces so let me try again from a fresh box and see what's up.

**Comment by dannylamb** [comment]
Gah, it should be `condition`, not `context`.  I made that error when testing and somehow it snuck in. One too many undo's...

**Comment by dannylamb** [comment]
`islandora_defaults` PR is up ^^

**Comment by kayakr** [comment]
@dannylamb Thanks @dannylamb, working for me.

**Comment by whikloj** [comment]
Testing
